### PR TITLE
feat: add config to specify custom urls in ogc-api-records

### PR DESCRIPTION
**Pull request by f-necas**
Custom urls can be specified with two config variables : 

- `linkToCustomMetadataUrl` boolean : Use custom url instead of legacyUrl or baseUrl
- `customMetadataUrl` string : url of the service. If none provided, url will be `/{metadata-uuid}`

### CODE REVIEW

File: modules/library/common-index-model/src/main/java/org/fao/geonet/index/converter/FormatterConfigurationImpl.java [hunk 1]
```diff
@@ -40,6 +49,11 @@ public String getSourceSiteTitle() {
 
   @Override
   public String buildLandingPageLink(String metadataId) {
+    if (linkToCustomMetadataUrl) {
+      return String.format("%s/%s",
+          customMetadataUrl,
```
Comment (pmauduit) [inline modules/library/common-index-model/src/main/java/org/fao/geonet/index/converter/FormatterConfigurationImpl.java:54]: you might need to document that the customMetadataUrl should not end with a final '/'. Also maybe a customMetadataUrlTemplate would make more sense as a variable name ?

**Review by pmauduit — APPROVED** [review]
Apart from one remark, The PR looks good but I don't feel I am legitimate to either accept / merge or deny it.

File: modules/library/common-index-model/src/main/java/org/fao/geonet/index/converter/FormatterConfigurationImpl.java (hunk 1 above)
Comment (pmauduit) [reply modules/library/common-index-model/src/main/java/org/fao/geonet/index/converter/FormatterConfigurationImpl.java:54]: > Also maybe a customMetadataUrlTemplate would make more sense as a variable name ?

sorry I misread your first comment in the PR, I thought that we could provide a url with a `{metadata-uuid}` which would be replaced, hence the "template" in the variable name, but reading the code, this is not the intent. You can forget about this part of my comment.

File: modules/library/common-search/src/test/java/org/fao/geonet/common/search/processor/impl/RssResponseProcessorImplTest.java [hunk 2]
```diff
@@ -120,6 +141,8 @@ public static PropertySourcesPlaceholderConfigurer properties() throws Exception
 			properties.setProperty("gn.site.organization", "momorg");
 			properties.setProperty("gn.baseurl", "http://gn5:8277/geonetwork");
 			properties.setProperty("gn.linkToLegacyGN4", "true");
+			properties.setProperty("gn.linkToCustomMetadataUrl", "false");
+      properties.setProperty("gn.customMetadataUrl", "http://gn:8277/custom");
```
Comment (jahow) [inline modules/library/common-search/src/test/java/org/fao/geonet/common/search/processor/impl/RssResponseProcessorImplTest.java:145]: looks like the file is using tabs and your changes are using spaces... ah, the good old times ;)

**Review by jahow — COMMENTED** [review]
I agree with @pmauduit, this option should be more documented, especially the trailing slash part (it is very common to specify urls with training slashes).

Other than a few old-fashioned space/tab mixing, everything looks good to me as well :) I can merge once @f-necas considers the PR done.

**Comment by jahow** [comment]
Ci is broken but that does not look related to your changes, merging.

Thanks a lot @f-necas, congrats for the first contribution here :tada:

**Pull request merged**